### ISSUE TITLE
"Polygon is self intersecting" error

### ISSUE THREAD

**Issue by gilboonet** [jscad/OpenJSCAD.org#536]
I use a web app that asks for a 3d model and a jscad script and creates on the fly a .jscad where the 3d model is used on the jscad script. It works fine, but I'm having this "polygon is self intersecting" that pops up when with the script that slices a volume into an interlocked 2 axis matrix.

Are there some tips that could help avoid this error ? Do you think such code can now be written with V2 or is it too early ?

## Expected Behavior
returns 6 interlocked slices

## Actual Behavior
Error : Polygon is self intersecting

## Steps to Reproduce the Problem

  1. Open https://openjscad.org/#https://raw.githubusercontent.com/gilboonet/gilboonet.github.io/master/demos/test_tr2.jscad
  2. Enter 75 as Scale (parameter "Echelle")
  3. Enter 0 as X rotation angle (parameter "Rotation X")

(it works with 1 as X rotation angle)

## Specifications

  - Version: 1.9.0 (current web version)
  - Platform: Ubuntu 19.10 64 bits
  - Environment: Firefox 74 (64 bits)

**Comment by z3dev** [comment]
I'm not sure why zero(0) for X rotation produces the error.

I noticed that you're using polygon() to build the larger pieces. Possible one of the polygons is empty, or resulting in a polygon with zero area. You might want to check those.

**Comment by gilboonet** [comment]
Thank you for your reply. I've filtered the polygons, in order to only use those having area > 1, and I tried to use fixed precision coordinates as I this project produces pattern that will be laser cut, millimeter precision is enough. I've kind of solve it by providing manually the coordinates of the slices, so now it works with 0 as X rotation angle. I'm hoping to be able to convert this project to V2.
